handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1103,11 +1101,14 @@ private SSLEngineResult sslReadErrorResult(int err, int bytesConsumed, int bytes
             if (handshakeException == null && handshakeState != HandshakeState.FINISHED) {
                 // we seems to have data left that needs to be transfered and so the user needs
                 // call wrap(...). Store the error so we can pick it up later.
-                handshakeException = new SSLHandshakeException(errStr);
+                handshakeException = new SSLHandshakeException(SSL.getErrorString(err));
             }
+            // We need to clear all errors so we not pick up anything that was left on the stack on the next
+            // operation. Note that shutdownWithError(...) will cleanup the stack as well so its only needed here.
+            SSL.clearError();
```

Comment:
@normanmaurer without this what behaviour did you observe?

Reply:
basically whatever test that ran after `OpenSslEngineTest.testMultipleRecordsInOneBufferWithNonZeroPositionJDKCompatabilityModeOff` failed when using BoringSSL. Thats also why I added asserts now to ensure the error stack is empty when we teardown.